Libraries/Movement/Movement.cpp

```diff
@@ -22,26 +22,25 @@ void Movement::advancePID(const double desire) {
   encoder_count_right_ = 0;
 
   do {
+    const double errorBNO = control_->getAngleError(bno_->getAngleX(), desire);
     control_->getPwmBNO(desire, pwm_left_final_bno, pwm_right_final_bno); 
     control_->getPwmUltrasonic(pwm_left_final_ultrasonic, pwm_right_final_ultrasonic);
 
-    double pwm_right_enginees = pwm_right_final_bno + pwm_right_final_ultrasonic; // Negative.
-    double pwm_left_enginees = pwm_left_final_bno + pwm_left_final_ultrasonic;    // Positive.
-
-    if (pwm_left_enginees > 0) {
-      pwm_right_enginees = Common::kLimitInfPwm;
-      control_->getPwm(pwm_left_enginees);
-    } else if (pwm_right_enginees < 0) {
-      pwm_right_enginees = -(pwm_right_enginees);
-      pwm_left_enginees = Common::kLimitInfPwm;
-      control_->getPwm(pwm_right_enginees);
+    if (errorBNO > 0) {
+      //pwm_right_final_bno += pwm_right_final_ultrasonic; 
+      control_->getPwm(pwm_right_final_bno);
     }
-    robot_->forwardPwm(pwm_right_enginees, pwm_left_enginees);
-  } while (encoder_count_left_ < kUnitLimit && encoder_count_right_ < kUnitLimit);
+    else {
+     // pwm_left_final_bno += pwm_left_final_ultrasonic;
+      control_->getPwm(pwm_left_final_bno);
+    }
+    robot_->forwardPwm(pwm_left_final_bno, pwm_right_final_bno);
+  } while (encoder_count_right_ < kUnitLimit && encoder_count_left_ < kUnitLimit);
+  robot_->stopEngines();
 }
 
 // TODO(MarlonB500): Implement the correct values for encoders.
-void Movement::advancePIDSwitches(const double desire) {
+/*void Movement::advancePIDSwitches(const double desire) {
```

Comment:
Update the TODO comment, since I guess now it's more than just the encoder value